src/context.rs

```diff
@@ -105,10 +106,83 @@ fn test_should_add_a_service() {
         }
     }
 
-    let service = ServiceStub;
-    let mut foo = Context::new(false, Some("localhost".to_owned()), None, None);
+    mod add_service {
+        use super::super::*;
 
-    assert_eq!(foo.services.is_empty(), true);
-    foo.add_service(Box::new(service));
-    assert_eq!(foo.services.is_empty(), false);
+        #[test]
+        fn test_should_increase_number_of_services() {
+            let service = super::ServiceStub;
+            let context = Context::shared(false, Some("localhost".to_owned()), None, None);
+            let mut locked_context = context.lock().unwrap();
```

Comment:
FYI, the macro approach gives another type of error:
```
src/context.rs:45:39: 45:40 error: unresolved name `shared_context` [E0425]
src/context.rs:45         setup!(service, shared_context, locked_context);
                                                        ^~
src/context.rs:45:9: 45:57 note: in this expansion of setup! (defined in src/context.rs)
src/context.rs:45:39: 45:40 help: run `rustc --explain E0425` to see a detailed explanation
src/context.rs:46:17: 46:40 error: the type of this value must be known in this context
src/context.rs:46         assert!(locked_context.services.is_empty());
                                  ^~~~~~~~~~~~~~~~~~~~~~~
src/context.rs:46:9: 46:53 note: in this expansion of assert! (defined in <std macros>)
````

 I isolated the [code that triggers the failure](https://github.com/JohanLorenzo/foxbox/blob/debug-setup-macro/src/context.rs#L45)